src/pkg/leaderElection.go

```diff
@@ -43,8 +44,34 @@ func RunLeaderElection(client *clientset.Clientset, lockName, lockIdentity, lock
 				log.Info().Msgf("Perform Migration")
 				for {
 					log.Info().Msgf("leader is %s", lockIdentity)
-					log.Info().Msgf("Getting replica count...")
-					time.Sleep(5 * time.Second)
+					replicaCount, err := getReplicas()
+					if err != nil {
+						log.Info().Msgf("Failed to get replica count: %v", err)
+						continue
+					}
+					log.Info().Msgf("Setting replica count to %v", replicaCount)
+					// Retry is being used below to prevent deployment update from overwriting a
+					// separate and unrelated update action per client-go's recommendation:
+					// https://github.com/kubernetes/client-go/blob/master/examples/create-update-delete-deployment/main.go#L117
+					retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+						deploymentsClient := client.AppsV1().Deployments(lockNamespace)
+						result, getErr := deploymentsClient.Get(context.TODO(), lockName, metav1.GetOptions{})
+						if getErr != nil {
+							log.Info().Msgf("Failed to get latest version of Deployment: %v", getErr)
+							return getErr
+						}
+						result.Spec.Replicas = &replicaCount
+						_, updateErr := deploymentsClient.Update(context.TODO(), result, metav1.UpdateOptions{})
+						return updateErr
+					})
+					if retryErr != nil {
+						log.Info().Msgf("Failed to set replica count: %v", retryErr)
+						continue
+					}
+					log.Info().Msgf("Successfully set replicas to %v", replicaCount)
+					// Not allowing this sleep interval to be configurable for now to prevent this value being set too low and
+					// calling the getReplicas API endpoint too frequently
+					time.Sleep(60 * time.Second)
```

Comment:
Because of the `continue`'s we now need to move this sleep to the top of the for loop so that it's always hit.  Otherwise if we ever hit a `continue` we'll loop immediately without the sleep.  This could cause a thundering herd effect if the error we are hitting persists every time.